VMEncryption/main/EncryptionConfig.py

```diff
@@ -73,6 +73,14 @@ def clear_config(self):
                 os.rename(self.encryption_environment.encryption_config_file_path, new_name)
             else:
                 self.logger.log(msg=("the config file not exist: {0}".format(self.encryption_environment.encryption_config_file_path)), level = CommonVariables.WarningLevel)
+            if clear_parameter_file:
+                if os.path.exists(self.encryption_environment.extension_parameter_file_path):
+                    self.logger.log(msg="archiving the encryption parameter file: {0}".format(self.encryption_environment.extension_parameter_file_path))
+                    time_stamp = datetime.datetime.now()
+                    new_name = "{0}_{1}".format(self.encryption_environment.extension_parameter_file_path, time_stamp)
+                    os.rename(self.encryption_environment.extension_parameter_file_path, new_name)
+                else:
+                    self.logger.log(msg=("the parameter file not exist: {0}".format(self.encryption_environment.extension_parameter_file_path)), level = CommonVariables.WarningLevel)
```

Comment:
Check for null before dereferencing self.encryption_environment.extension_parameter_file_path ? 
If parameter file does not exist,  should this log message be informational rather than a warning?  If the file does not exist then it is already "cleared" and this wouldn't be a problem.